cob_twist_controller/src/cob_twist_controller.cpp

```diff
@@ -171,24 +185,60 @@ bool CobTwistController::initialize()
     return true;
 }
 
+void CobTwistController::reinitServiceRegistration()
```

Comment:
could you not register a vector or links?
and then manage a "vector" of obstacle distance calculations?

(or rather a map with key collision_check_frames